RealmSwift-swift1.2/RealmConfiguration.swift

```diff
@@ -0,0 +1,156 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import Realm
+import Realm.Private
+
+/**
+A `RealmConfiguration` is used to describe the different options used to
+create a `Realm` instance.
+*/
+public struct RealmConfiguration {
```

Comment:
You haven't elaborated on why `Configuration` is too generic in the Swift API only, nor on it is special in that regard. Is `Configuration` too generic because it's possible we'll want to make some other object in RealmSwift configurable in the future? If so, why wouldn't we also give the Objective-C class a name that indicates that it is a configuration for a Realm (e.g., `RLMRealmConfiguration`). Is there something I'm missing that makes the need to qualify the typename specific to the Swift API? My impression is that either both should have the extra `Realm` qualification, or neither should.